src/Plugins/ComplexCore/src/ComplexCore/Filters/RawBinaryReaderFilter.cpp

```diff
@@ -0,0 +1,203 @@
+  params.insert(std::make_unique<UInt64Parameter>(k_SkipHeaderBytes_Key, "Skip Header Bytes", "", 0));
+  params.insert(std::make_unique<ArrayCreationParameter>(k_CreatedAttributeArrayPath_Key, "Output Attribute Array", "", DataPath(std::vector<std::string>{"Imported Array"})));
+
+  return params;
+}
+
+//------------------------------------------------------------------------------
+IFilter::UniquePointer RawBinaryReaderFilter::clone() const
+{
+  return std::make_unique<RawBinaryReaderFilter>();
+}
+
+//------------------------------------------------------------------------------
+IFilter::PreflightResult RawBinaryReaderFilter::preflightImpl(const DataStructure& dataStructure, const Arguments& filterArgs, const MessageHandler& messageHandler) const
+{
+  auto pInputFileValue = filterArgs.value<FileSystemPathParameter::ValueType>(k_InputFile_Key);
+  auto pScalarTypeValue = filterArgs.value<NumericType>(k_ScalarType_Key);
+  auto pNumberOfComponentsValue = filterArgs.value<int32>(k_NumberOfComponents_Key);
+  auto pSkipHeaderBytesValue = filterArgs.value<uint64>(k_SkipHeaderBytes_Key);
+  auto pCreatedAttributeArrayPathValue = filterArgs.value<DataPath>(k_CreatedAttributeArrayPath_Key);
+
+  if(pNumberOfComponentsValue < 1)
+  {
+    auto error = MakeErrorResult<OutputActions>(k_RbrZeroComponentsError, "The number of components must be positive.");
+    return {std::move(error), {}};
+  }
+
+  complex::Result<OutputActions> resultOutputActions;
+
+  uint32 inputFileSize = std::filesystem::file_size(pInputFileValue);
+  if(inputFileSize == 0)
+  {
+    auto error = MakeErrorResult<OutputActions>(k_RbrEmptyFile, "File '" + pInputFileValue.string() + "' is empty.");
+    return {std::move(error), {}};
+  }
+
+  uint32 intendedFileSize = inputFileSize - pSkipHeaderBytesValue;
+  if(intendedFileSize <= 0)
+  {
+    auto error = MakeErrorResult<OutputActions>(k_RbrSkippedTooMuch, "More bytes have been skipped than exist in file '" + pInputFileValue.string() + "'.");
+    return {std::move(error), {}};
+  }
+
+  uint32 typeSize = 0;
+  switch(pScalarTypeValue)
+  {
+  case NumericType::int8:
+    typeSize = sizeof(int8);
+    break;
+  case NumericType::uint8:
+    typeSize = sizeof(uint8);
+    break;
+  case NumericType::int16:
+    typeSize = sizeof(int16);
+    break;
+  case NumericType::uint16:
+    typeSize = sizeof(uint16);
+    break;
+  case NumericType::int32:
+    typeSize = sizeof(int32);
+    break;
+  case NumericType::uint32:
+    typeSize = sizeof(uint32);
+    break;
+  case NumericType::int64:
+    typeSize = sizeof(int64);
+    break;
+  case NumericType::uint64:
+    typeSize = sizeof(uint64);
+    break;
+  case NumericType::float32:
+    typeSize = sizeof(float32);
+    break;
+  case NumericType::float64:
+    typeSize = sizeof(float64);
+    break;
+    //  case NumericType::boolean:
+    //    break;
+  default:
+    throw std::runtime_error("The chosen scalar type is not supported by this filter.");
+  }
```

Comment:
We could probably make this into a utility function in `TypesUtility.hpp`. Something like
```cpp
template <class T>
inline constexpr usize GetNumericTypeSize();
```